kgo.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const foreign = require('foreign');
+const kgo = require('kgo');
+const R = require('ramda');
+const S = require('sanctuary');
+const shuv = require('shuv');
+
+
+const readFile = (filename, callback) => fs.readFile(filename, {encoding: 'utf8'}, callback);
+const complete = (error, result) => {
+  if(error == null){
+    process.stdout.write(result);
+    process.exit(0);
+  }else{
+    process.stderr.write(String(error) + '\n');
+    process.exit(1);
+  }
+};
+
+
+const main = () => {
+  const relative = filename => path.join(process.argv[2], filename);
+
+  kgo
+  ('index', shuv(readFile, relative('index.txt')))
+  ('filePaths', ['index'], kgo.sync(index => S.lines(index).map(relative)))
+  ('files', ['filePaths'], shuv(foreign.parallel, readFile))
+  ('concated', ['files'], kgo.sync(R.join('')))
+  (['*', 'concated'], complete);
```

Comment:
'*' refers to the first error result from any task.

so that last task run `complete` if any error happens, OR `concated` completes without error.